api/client/service.proto

```diff
@@ -34,10 +36,59 @@ message ExecutionResponse {
   google.rpc.Status error_detail = 7;
 }
 
+// Used to request a RequestStreamResponse.
+message RequestStreamRequest {
+  // Amount of RequestStreamResponses that should be yielded.
+  // Used to implement basic flow control by the client. Once AsyncClientImpl gets
+  // a way to apply backpressure this could probably be dropped.
+  uint64 amount = 1;
+}
+
+message RequestStreamResponse {
+  // Specifies what the requests look like. Will be merged with the client's configuration.
+  RequestSpecifier request_specifier = 1;
+  // Response-level expectations associated to the above request specification.
+  Expectations expectations = 2;
+}
+
+message RequestSpecifier {
+  // method/path/authority/content_length can all be expressed in request headers.
+  // so technically they are redundant. In case of ambiguity, the explicit fields will prevail.
+  // Note that as of today, the target upstream/cluster that will be used to replay traffic to
+  // fixes the destination address/port.
+  // Request method
+  google.protobuf.StringValue method = 1;
+  // Request path
+  google.protobuf.StringValue path = 2;
+  // Request authority
+  google.protobuf.StringValue authority = 3;
+  // Request content length. The client will transfer the number of bytes specified here for the
+  // request body.
+  google.protobuf.UInt32Value content_length = 4;
+  // Request header replacements. Any existing header(s) with the same name will be removed
+  // before setting.
+  envoy.api.v2.core.HeaderMap headers = 5;
```

Comment:
Considering this spec is likely to become a source of truth for people to refer to, would it be possible to link to a reference of what headers are supported by envoy/NH backend?  Does such a list exist somewhere?

Reply:
Well, I guess this can take arbitrary string/string combinations. But there's quite a few standardized and semi-standardized headers, e.g.: https://en.wikipedia.org/wiki/List_of_HTTP_header_fields

Reply:
My initial comment was really unclear, and I'm sorry.  I did some more digging to explain what I really mean. What I thought would be helpful here would be to callout in a comment some pointer to documentation on x-nighthawk-test-server-config which I think has its options defined here: https://github.com/envoyproxy/nighthawk/blob/master/api/server/response_options.proto, so that users could understand what might be useful to specify here.  However, I now see that content_length is currently the only variable controlled by nighthawk-specific headers.  Is there an idea that we expect to implement more features like that?  If not, probably no need to call it out here, because you've abstracted that behavior away from the headers anyway

Reply:
Actually, looking again, I think that the content_length specified about is request length, not response length, so I do think providing a pointer to response_options.proto would probably be helpful, if you agree.

Reply:
Oh I see, that would be when we're configured to send these request headers to our own test server, which may interpret some of the request headers as configuration directives. In that case there's quite a bit of inception going on when we talk about that here :-) To be honest, I'm hesitant about documenting here because it is not written in stone that we'll be talking to the test server, and because of the inception it might take a lot of stack space to grok what we're talking about? 
I think we need some more high level documentation / cookbooks / getting started. There we could document
setups and describe the flow of what happens where, and this could fit more naturally. WDYT?

Reply:
That's a really good point. I wasn't thinking about other consumers who may not be using the NH Test Server.  Having a recipe book of sorts, some high level documentation would be really awesome.  I'm fine with omitting it here in that case.